app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/hooks/useProjectManagerAdapter.ts

```diff
@@ -0,0 +1,122 @@
+/** Implement the project manager desktop side adapter(electron) */
+const useProjectManagerDesktopClient = (
+    projectManager?: ProjectManager
+): ProjectManagerAdapter | null => {
+    if (!projectManager) return null
+    const { organization } = auth.useFullUserSession()
+    const baseOnEmptyProject = (
+        project: Partial<Project> & Pick<Project, 'organizationId' | 'name' | 'projectId'>
+    ): Project => ({
+        state: { type: ProjectState.created },
+        packageName: 'Main',
+        address: null,
+        ami: null,
+        ideVersion: null,
+        engineVersion: null,
+        ...project,
+    })
+    return {
+        async createProject(projectName, projectTemplateName) {
+            const createdProject = (await projectManager.createProject(
+                projectName,
+                projectTemplateName
+            )) as CreateProjectResult
+            const project = baseOnEmptyProject({
+                organizationId: organization.id,
+                projectId: createdProject.result.projectId,
+                name: createdProject.result.projectName,
+            })
+            return project
+        },
+        async listProjects() {
+            const localProjects = ((await projectManager.listProjects()) as ListProjectResult)
+                .result.projects
+            const projects = localProjects.map(p =>
+                baseOnEmptyProject({
+                    organizationId: organization.id,
+                    projectId: p.id,
+                    name: p.name,
+                })
+            )
+            return projects
+        },
+    }
+}
+
+// === useProjectManagerWebClient ===
+/** Implement the project manager web side adapter(web) */
+const useProjectManagerWebClient = (): ProjectManagerAdapter => {
+    const backendService = useBackendService()
+    return {
+        async createProject(projectName, projectTemplateName) {
+            const project = await backendService.createProject({
+                projectName,
+                projectTemplateName: projectTemplateName?.toLowerCase(),
+            })
+            return project
+        },
+        async listProjects() {
+            return await backendService.listProjects()
+        },
+    }
+}
+
+/** the useProjectManageAdapter options */
+interface UseProjectManageAdapterOptions {
+    runningOnDesktop: boolean
+    projectManager?: ProjectManager
+}
+
+/** A React hook to use the adapter of the project manager.
+ *
+ * The purpose of this hook is to resolve platform differences in requests to the projectManager.
+ */
+const useProjectManageAdapter = (
+    options: UseProjectManageAdapterOptions
+): ProjectManagerAdapter => {
+    const { runningOnDesktop, projectManager } = options
+    const projectManagerDesktopClient = useProjectManagerDesktopClient(projectManager)
+    const projectManagerWebClient = useProjectManagerWebClient()
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
i'd either add a comment why it's safe to do the non-null assertion, or remove the comment so that it stays as a warning, so that it's obvious that it's an issue that will need to be fixed later